hotshot-state-prover/src/bin/gen-demo-contract.rs

```diff
@@ -0,0 +1,157 @@
+    ///          In the future we should get the stake table from the contract.
+    #[clap(
+        short,
+        long,
+        env = "ESPRESSO_ORCHESTRATOR_NUM_NODES",
+        default_value = "5"
+    )]
+    num_nodes: usize,
+
+    /// Seed to use for generating node keys.
+    /// WARNING: This is used temporarily to initialize a static stake table.
+    ///          In the future we should get the stake table from the contract.
+    #[arg(long, env = "ESPRESSO_ORCHESTRATOR_KEYGEN_SEED", default_value = "0x0000000000000000000000000000000000000000000000000000000000000000", value_parser = parse_seed)]
+    keygen_seed: [u8; 32],
+
+    #[clap(long, default_value = "contracts")]
+    contracts_root_dir: PathBuf,
+}
+
+#[derive(Debug, Snafu, From)]
+enum ParseSeedError {
+    #[snafu(display("seed must be valid hex: {source}"))]
+    Hex { source: FromHexError },
+
+    #[snafu(display("wrong length for seed {length} (expected 32)"))]
+    WrongLength { length: usize },
+}
+
+fn parse_seed(s: &str) -> Result<[u8; 32], ParseSeedError> {
+    <[u8; 32]>::try_from(hex::decode(s)?)
+        .map_err(|vec| ParseSeedError::WrongLength { length: vec.len() })
+}
+
+pub fn stake_table_commitment_for_demo(
+    num_nodes: usize,
+    keygen_seed: [u8; 32],
+) -> ((CircuitField, CircuitField, CircuitField), U256) {
+    // We now initialize a static stake table as what hotshot orchestrator does.
+    // In the future we should get the stake table from the contract.
+    let mut st = StakeTable::<BLSPubKey, StateVerKey, CircuitField>::new(STAKE_TABLE_CAPACITY);
+    (0..num_nodes).for_each(|id| {
+        let bls_key = BLSPubKey::generated_from_seed_indexed(keygen_seed, id as u64).0;
+        let state_ver_key =
+            StateKeyPair::generate_from_seed_indexed(keygen_seed, id as u64).ver_key();
+        st.register(bls_key, U256::from(1u64), state_ver_key)
+            .expect("Key registration shouldn't fail.");
+    });
+    st.advance();
+    st.advance();
+    (
+        st.commitment(SnapshotVersion::LastEpochStart)
+            .expect("Commitment computation shouldn't fail."),
+        st.total_stake(SnapshotVersion::LastEpochStart).unwrap() * 2 / 3,
+    )
+}
+
+fn main() {
+    let args = Args::parse();
+
+    let ((bls_comm, schnorr_comm, stake_comm), threshold) =
+        stake_table_commitment_for_demo(args.num_nodes, args.keygen_seed);
+
+    // calculate the path to solidity file
+    let contract_name = "LightClient.s.sol";
+    let mut path = args.contracts_root_dir;
+    path.push("script");
+    path.push(contract_name);
+    path.set_extension("sol");
+
+    println!(
+        "Writing contract into file: {}",
+        path.as_os_str().to_str().unwrap()
+    );
+    // overwrite the file
+    let mut file = OpenOptions::new()
+        .write(true)
+        .create(true)
+        .truncate(true)
+        .open(path)
+        .unwrap();
+
+    let code = format!(
+        "// SPDX-License-Identifier: UNLICENSED
```

Comment:
Add a comment about the fact that this code is generated automatically.

Reply:
added here [73e8fb4](https://github.com/EspressoSystems/espresso-sequencer/pull/1041/commits/73e8fb4b637e01ecc045b21cfa7c85463117d3a9)